src/measure_theory/l1_space.lean

```diff
@@ -16,24 +58,72 @@ set_option class.instance_max_depth 100
 namespace measure_theory
 open set lattice filter topological_space ennreal emetric
 
-universes u v
-variables {α : Type u} {β : Type v} [measure_space α]
-variables {γ : Type*} [normed_group γ]
+universes u v w
+variables {α : Type u} [measure_space α]
+variables {β : Type v} [normed_group β]
+variables {γ : Type w} [normed_group γ]
 
-infixr ` →ₘ `:25 := ae_eq_fun
+def integrable (f : α → β) : Prop := (∫⁻ a, nnnorm (f a)) < ⊤
 
-def integrable (f : α → γ) : Prop := (∫⁻ a, nnnorm (f a)) < ⊤
+lemma integrable_of_ae_eq {f g : α → β} (hf : integrable f) (h : ∀ₘ a, f a = g a) : integrable g :=
+begin
+  simp only [integrable] at *,
+  have : (∫⁻ (a : α), ↑(nnnorm (f a))) = (∫⁻ (a : α), ↑(nnnorm (g a))),
+    apply lintegral_congr_ae,
+    filter_upwards [h],
+    assume a,
+    simp only [mem_set_of_eq],
+    assume h,
+    rw h,
+  rwa ← this
+end
 
-@[simp] lemma lintegral_nnnorm_zero : (∫⁻ a : α, nnnorm (0 : γ)) = 0 := by simp
+lemma integrable_iff_of_ae_eq (f g : α → β) (h : ∀ₘ a, f a = g a) : integrable f ↔ integrable g :=
+iff.intro (λhf, integrable_of_ae_eq hf h) (λhg, integrable_of_ae_eq hg (all_ae_eq_symm h))
 
-lemma integrable_zero : integrable (0 : α → γ) :=
+lemma lintegral_nnnorm_eq_lintegral_edist (f : α → β) :
+  (∫⁻ a, nnnorm (f a)) = ∫⁻ a, edist (f a) 0 :=
+begin
+  apply lintegral_congr_ae,
+  filter_upwards [],
+  assume a,
+  simp only [mem_set_of_eq],
+  rw [edist_nndist, nndist_eq_nnnorm, sub_zero (f a)]
+end
+
+lemma integrable_iff_lintegral_edist (f : α → β) :
+  integrable f ↔ (∫⁻ a, edist (f a) 0) < ⊤ :=
+by rw [integrable, lintegral_nnnorm_eq_lintegral_edist]
+
+lemma lintegral_edist_triangle [second_countable_topology β] {f g h : α → β}
+  (hf : measurable f) (hg : measurable g) (hh : measurable h) :
```

Comment:
you just need `h` to be measurable here: then, if `f` or `g` is not measurable, the right hand side is top.

Reply:
Thanks for the review! But this one I am not so sure. It seems to me that the integral of a non-measurable function is not necessarily top. Here is the definition of `lintegral`:
```
def lintegral (f : α → ennreal) : ennreal :=
⨆ (s : α →ₛ ennreal) (hf : f ≥ s), s.integral
```

Reply:
You are absolutely right!